### PR TITLE
Virt net modify dns configuration

### PR DESCRIPTION
**Pull request by torvitas**
##### SUMMARY
Enable modification of dns entries.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
virt_net

### CODE REVIEW

**Comment by torvitas** [comment]
ready_for_review

**Comment by Akasurde** [comment]
cc @goneri

**Comment by Akasurde** [comment]
@torvitas Thanks for the contribution. PR https://github.com/ansible/ansible/pull/53276 adds unit test for `virt_net` module. If you don't mind we can wait for this PR to merge first and add a unit test for this change according to #53276.  Let me know if you have concern. Thanks.

**Comment by torvitas** [comment]
I am fine with that.

**Comment by Akasurde** [comment]
@torvitas Great. Thanks for the help.

File: lib/ansible/modules/cloud/misc/virt_net.py [hunk 1]
```diff
@@ -273,8 +279,46 @@ def modify(self, entryid, xml):
...
+                    return False
+                else:
+                    if not self.module.check_mode:
+                        res = 0
+                        res += network.update(libvirt.VIR_NETWORK_UPDATE_COMMAND_DELETE,
+                                              libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+                                              -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
+                        res += network.update(libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
+                                              libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+                                              -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
```
Comment (goneri) [inline lib/ansible/modules/cloud/misc/virt_net.py]: Thanks for the patch. It looks pretty good, I'm just not a big fan of the use of `libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT` here and line 296. If the user restarts the network, the change will be gone and I don't think this is what is expected. You can `libvirt.VIR_DOMAIN_AFFECT_LIVE|libvirt.VIR_DOMAIN_AFFECT_CONFIG` instead.

**Review by goneri — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/misc/virt_net.py (hunk 1 above)
Comment (torvitas) [reply lib/ansible/modules/cloud/misc/virt_net.py]: It's done the way it's done for dhcp. And I didn't think it actually works as documented.

VIR_NETWORK_UPDATE_AFFECT_CURRENT | = | 0 (0x0) | affect live if network is active, config if it's not active
-- | -- | -- | --

That doesn't sound very reasonable to me.

Comment (goneri) [reply lib/ansible/modules/cloud/misc/virt_net.py]: I agree. `VIR_NETWORK_UPDATE_AFFECT_CURRENT` depends on the state of the resource. This is not what an Ansible user expects IMHO. Thanks for the fix (7698fadb176587e7f4272f985aac9dc09a8a2b63).

**Comment by torvitas** [comment]
Is there anything I can do to make progress on this?

**Review by goneri — APPROVED** [review]

**Comment by goneri** [comment]
@torvitas could you add a test coverage in ./test/integration/targets/virt_net?

**Comment by torvitas** [comment]
sounds reasonable

**Comment by Akasurde** [comment]
@torvitas Could you please fix this - 
```
00:58 ERROR: Found 1 pep8 issue(s) which need to be resolved:
00:58 ERROR: lib/ansible/modules/cloud/misc/virt_net.py:301:27: E111 indentation is not a multiple of four (100%)
```
You can find more details here - https://app.shippable.com/github/ansible/ansible/runs/123907/1/console

Thanks.

**Comment by goneri** [comment]
Hi @torvitas, I don't see any change of the functional tests. Is this something you still plan to do?

**Comment by torvitas** [comment]
> Hi @torvitas, I don't see any change of the functional tests. Is this something you still plan to do?
I defenitly plan to delivery these tests. Right now I am just busy with another part of the project, but I will come back to this.

**Comment by goneri** [comment]
> > Hi @torvitas, I don't see any change of the functional tests. Is this something you still plan to do?
> 
> I defenitly plan to delivery these tests. Right now I am just busy with another part of the project, but I will come back to this.

Thank you @torvitas for that :-)

**Pull request closed without merging**